lib/src/exceptions.dart

```diff
@@ -118,3 +119,34 @@ bool isUserFacingException(error) {
       error is YamlException ||
       error is UsageException;
 }
+
+/// An exception thrown when parsing a pubspec.yaml or a pubspec.lock.
+///
+/// These exceptions are often thrown lazily while accessing pubspec properties.
```

Comment:
Not really, we explicitly call `allErrors` :D

But yes, it makes sense that they are lazy, because we have to read the SDK constraint before we care if there are other errors.

Reply:
I think we only call that during publication.